src/lib/typenames.ts

```diff
@@ -72,6 +72,19 @@ function addTypenameToDocument(doc: DocumentNode) {
 // Adds __typename fields to a GraphQL query string
 export function formatTypeNames(query: string): string {
   const doc = parse(query);
+
+  const hasSubscriptionQuery = doc.definitions.some(
+    (definition: OperationDefinitionNode) =>
+      definition.operation === 'subscription'
+  );
+
+  // Subscriptions aren't allowed to query for more than one of anything per the spec.
+  // @see https://facebook.github.io/graphql/draft/#sec-Single-root-field
+  // This prevents a subscription from getting a typename injection which would cause the above error
+  if (hasSubscriptionQuery) {
+    return print(doc);
```

Comment:
We probably still want to add __typename to anything but the root query fiel set though, since itd be hard to implement solid and more complex caching strategies otherwise

Reply:
word. I'll try and figure that out.